dex/src/state.rs

```diff
@@ -2686,8 +2691,201 @@ impl State {
     }
 
     #[cfg(feature = "program")]
-    fn process_send_take(_args: account_parser::SendTakeArgs) -> DexResult {
-        unimplemented!()
+    fn process_send_take(args: account_parser::SendTakeArgs) -> DexResult {
+
+        let account_parser::SendTakeArgs {
+            instruction,
+            signer,
+            mut event_q,
+            mut req_q,
+            mut order_book_state,
+            coin_wallet,
+            pc_wallet,
+            coin_vault,
+            pc_vault,
+            spl_token_program,
+            fee_tier,
+            vault_signer
+        } = args;
+
+        // Question about architecture of max coin, max price, min, etc
+        let limit_price = instruction.limit_price;
+        let mut limit = instruction.limit;
+        let min_coin_qty = instruction.min_coin_qty;
+        let min_native_pc_qty = instruction.min_native_pc_qty;
+
+        let order_type = OrderType::ImmediateOrCancel;
+        let self_trade_behavior = SelfTradeBehavior::AbortTransaction;
+        let native_pc_qty_locked;
+
+        let order_id = req_q.gen_order_id(instruction.limit_price.get(), instruction.side); 
+        let mut proceeds = RequestProceeds::zero();
+        
+        match instruction.side {
+            Side::Bid => {
+                let lock_qty_native = instruction.max_native_pc_qty_including_fees;
+                native_pc_qty_locked = Some(lock_qty_native);
+            }
+            Side::Ask => {
+                native_pc_qty_locked = None;
+            }
+        };
+
+        let request = RequestView::NewOrder {
+            side: instruction.side,
+            order_type,
+            order_id,
+            fee_tier,
+            self_trade_behavior,
+            // Pass dummy account - system program pubkey
+            owner: solana_program::system_program::ID.to_aligned_bytes(),
+            // Pass 0 
+            owner_slot: 0,
+            max_coin_qty: instruction.max_coin_qty,
+            native_pc_qty_locked,
+            client_order_id: None,
+        };
+
+        let _unfilled_portion = order_book_state.process_orderbook_request(
```

Comment:
Why is `unfilled_portion` unused? If FOK were properly implemented (in the case that `min_coin_qty == max_coin_qty`), you would need to check that this amount is equal to 0

Reply:
proceeds tracks the amount that is credited to / debited from account

Reply:
also, unfilled_portion should always return None, based on how logic in new_bid/new_ask works. Maybe a different name would be better in this case (because we aren't actually filling).

Reply:
Isn't it simpler to just check `unfilled_portion < max - min`

Reply:
But unfilled_portion will always be none, even if some of our order doesn't get filled

Reply:
Hm is this true? I need to look into the logic a little more. Also feels like the testing needs to be much more thorough

Reply:
I also don't see why it has to be always none